src/link_manager/LinkManagerStateMachine.cpp

```diff
@@ -974,10 +974,17 @@ void LinkManagerStateMachine::startMuxWaitTimer(uint32_t factor)
 void LinkManagerStateMachine::handleMuxWaitTimeout(boost::system::error_code errorCode)
 {
     if (errorCode == boost::system::errc::success) {
+        mMuxWaitTimeoutCount++;
         if (mMuxStateMachine.getWaitStateCause() == mux_state::WaitState::WaitStateCause::SwssUpdate) {
             MUXLOGTIMEOUT(mMuxPortConfig.getPortName(), "orchagent timed out responding to linkmgrd", mCompositeState);
         } else if (mMuxStateMachine.getWaitStateCause() == mux_state::WaitState::WaitStateCause::DriverUpdate) {
             MUXLOGTIMEOUT(mMuxPortConfig.getPortName(), "xcvrd timed out responding to linkmgrd", mCompositeState);
+            // on the 3rd timeout, send switch active command to peer
+            if (mMuxWaitTimeoutCount == mMuxPortConfig.getNegativeStateChangeRetryCount()) { 
```

Comment:
what is the getNegativeStateChangeRetryCount? will this create false positives?

Reply:
NegativeStateChangeRetryCount is a configurable threshold. It's 3 by default. Will increase the timeout limit.

Reply:
Updated MAX_BACKOFF_FACTOR to 32 in line 64. Now each timeout will be 9.6 seconds.